examples/generate_image_dataset/generate_poses.py

```diff
@@ -42,33 +42,28 @@
         # reset the simulation
         _ = env.reset()
 
-        # # get random points based on number of poses requested
-        # _, _, _, step_info = env.step(
-        #     action={
-        #         "set_pose": np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0], dtype=np.float32),
-        #         "set_num_random_points": np.array([args.num_poses_per_scene], dtype=np.uint32)})
-
-        # positions = step_info["agent_step_info"]["random_points"]
-
-        # # generate random pitch, yaw, roll values
-        # pitch_values = np.random.uniform(low=0.0, high=0.0, size=args.num_poses_per_scene)
-        # yaw_values = np.random.uniform(low=-180.0, high=180.0, size=args.num_poses_per_scene)
-        # roll_values = np.random.uniform(low=0.0, high=0.0, size=args.num_poses_per_scene)
-
-        # # store poses in a csv file
-        # df = pd.DataFrame({"scene_id"  : scene_id,
-        #                    "pos_x_cms" : positions[:,0],
-        #                    "pos_y_cms" : positions[:,1],
-        #                    "pos_z_cms" : positions[:,2] + config.SIMULATION_CONTROLLER.CAMERA_AGENT.NAVMESH.AGENT_HEIGHT,
-        #                    "pitch_degs": pitch_values,
-        #                    "yaw_degs"  : yaw_values,
-        #                    "roll_degs" : roll_values})
-        # df.to_csv(args.poses_file, mode="w" if i==0 else "a", index=False, header=i==0)
-
-        # plt.scatter(positions[:,0], positions[:,1], s=1.0)
-        # plt.gca().set_aspect("equal")
-        # plt.gca().invert_yaxis() # invert the y-axis so the plot matches a top-down view of the scene in Unreal
-        # plt.show()
+        # get a few random points
+        points = env.get_random_points(args.num_poses_per_scene)
+
+        # generate random pitch, yaw, roll values
+        pitch_values = np.random.uniform(low=0.0, high=0.0, size=args.num_poses_per_scene)
+        yaw_values   = np.random.uniform(low=-180.0, high=180.0, size=args.num_poses_per_scene)
+        roll_values  = np.random.uniform(low=0.0, high=0.0, size=args.num_poses_per_scene)
+
+        # store poses in a csv file
+        df = pd.DataFrame({"scene_id"            : scene_id,
+                           "location_x_cms"      : points[:,0],
+                           "location_y_cms"      : points[:,1],
+                           "location_z_cms"      : points[:,2] + config.SIMULATION_CONTROLLER.NAVMESH.AGENT_HEIGHT,
+                           "rotation_pitch_degs" : pitch_values,
+                           "rotation_yaw_degs"   : yaw_values,
+                           "rotation_roll_degs"  : roll_values})
+        df.to_csv(args.poses_file, mode="w" if i==0 else "a", index=False, header=i==0)
```

Comment:
I was never a fan of this approach for generating csv files. Just append to `df` and write it once. The current approach forces you to use `enumerate` and specify a bunch of extra named arguments to `to_csv`, which is slightly uglier than necessary.

Reply:
Updated